### PR TITLE
[Translation] [russian] - translates pointers into Russian

### PR DESCRIPTION
**Pull request by koleaby4**

### CODE REVIEW

File: _content/tour/rus/_en_ru_dictionary.yml [hunk 1]
```diff
@@ -35,9 +35,18 @@ Data-oriented design:
...
     ru: отладка программного обеспечения
 
+Escape analysis:
+    ru: анализ побега (памяти из стэка на кучу)
+
 Floating point number:
     ru: число с плавающей запятой
 
+Goroutine:
+    ru: Go-рутина
```
Comment (rumyantseva) [inline _content/tour/rus/_en_ru_dictionary.yml:45]: This one is tricky :) In Donovan-Kernighan's translation, it's called "Go-подпрограмма", but in the community, everyone calls it "горутина" :) So, I'd suggest "горутина".

File: _content/tour/rus/pointers.article [hunk 2]
```diff
@@ -1,151 +1,150 @@
...
- 
-Stacks are self cleaning and zero value helps with the initialization of the stack.
-Every time you make a function call, and a frame of memory is blocked out, the memory
-for that frame is initialized, which is how the stack is self cleaning. On a function
-return, the memory for the frame is left alone since it’s unknown if that memory
-will be needed again. It would be inefficient to initialize memory on returns.
+Размер фрейма для каждой функции определяется при компиляции. На стеке нельзя
+создать значение, если на этапе компиляции компилятор не знает размера этого значения.
+Если компилятор не знает размера значения на этапе компиляции, это значение должно
+быть создано в куче.
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:43]:
```suggestion
Размер фрейма для каждой функции определяется при компиляции. Значение не может быть
создано на стеке, если на этапе компиляции компилятор не знает размера этого значения.
Если компилятор не знает размера значения на этапе компиляции, значение вынужденно 
создается на куче.
```

File: _content/tour/rus/pointers.article [hunk 3]
```diff
@@ -1,151 +1,150 @@
...
+инициализируется, что позволяет самоочищение стека. При возврате из функции
+память фрейма остается нетронутой, поскольку неизвестно, потребуется ли эта память снова.
+Инициализация памяти при возврате была бы неэффективной.
 
-All data is moved around the program by value. This means as data is being passed
-across program boundaries, each function or Goroutine is given its own copy of the
-data. There are two types of data you will work with, the value itself (int,
-string, user) or the value's address. Addresses are data that need to be copied
-and stored across program boundaries.
+*Передача* *по* *Значению*
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:51]:
```suggestion
*Передача* *по* *значению*
```

File: _content/tour/rus/pointers.article [hunk 4]
```diff
@@ -1,151 +1,150 @@
...
 
-The following code attempts to explain this more.
+По ходу выполнения программы, все данные передаются по значению. Это означает, что при
+передаче данных через границы программы, каждая функция или горутина получают свою
+собственную копию данных. Существуют два типа данных, с которыми вы будете работать:
+само значение (int, string, пользователь) или адрес значения. Адреса - это данные,
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:56]:
```suggestion
само значение (int, string, user) или адрес значения. Адреса - это данные,
```

File: _content/tour/rus/pointers.article [hunk 5]
```diff
@@ -1,151 +1,150 @@
...
 
-        // Pass a copy of the "value of" count (what’s in the box)
-        // to the increment1 function.
+        // Передача копии "значения" `count` (то, что находится в коробке)
+        // функции increment1.
         increment1(count)
 
-        // Print out the "value of" and "address of" count.
-        // The value of count will not change after the function call.
+        // Вывод "значения" и "адреса" переменной счет.
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:73]:
```suggestion
        // Вывод "значения" и "адреса" переменной count.
```

File: _content/tour/rus/pointers.article [hunk 6]
```diff
@@ -1,151 +1,150 @@
...
+        // Значение счета не изменится после вызова функции.
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:74]:
```suggestion
        // Значение count не изменится после вызова функции.
```

File: _content/tour/rus/pointers.article [hunk 7]
```diff
@@ -1,151 +1,150 @@
...
         println("count:\tValue Of[", count, "]\tAddr Of[", &count, "]")
 
-        // Pass a copy of the "address of" count (where is the box)
-        // to the increment2 function. This is still considered a pass by
-        // value and not a pass by reference because addresses are values.
+        // Передача копии "адреса" счета (где находится коробка)
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:77]:
```suggestion
        // Передача копии "адреса" переменной count (где находится коробка)
```

File: _content/tour/rus/pointers.article [hunk 8]
```diff
@@ -1,151 +1,150 @@
...
+        // функции increment2. Это все равно считается передачей по значению,
+        // а не передачей по ссылке, потому что адреса - это значения.
         increment2(&count)
 
-        // Print out the "value of" and "address of" count.
-        // The value of count has changed after the function call.
+        // Вывод "значения" и "адреса" счета.
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:82]:
```suggestion
        // Вывод "значения" и "адреса" переменной count.
```

File: _content/tour/rus/pointers.article [hunk 9]
```diff
@@ -1,151 +1,150 @@
...
+        // Значение счета изменилось после вызова функции.
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:83]:
```suggestion
        // Значение переменной count изменилось после вызова функции.
```

File: _content/tour/rus/pointers.article [hunk 10]
```diff
@@ -1,151 +1,150 @@
...
-    // increment2 declares the function to accept its own copy of
-    // an address that points to an integer value.
-    // Pointer variables are literal types and are declared using *.
+    // increment2 объявляет функцию, принимающую копию адреса,
+    // указывающего на целочисленное значение.
+    // Переменные-указатели - это литеральные типы которые объявляются с помощью *.
     func increment2(inc *int) {
 
-        // Increment the caller’s int value through the pointer.
+        // Увеличить целочисленное значение вызывающего через указатель.
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:103]:
```suggestion
        // Увеличить целочисленное значение в вызывающей функции через указатель.
```

File: _content/tour/rus/pointers.article [hunk 11]
```diff
@@ -1,151 +1,150 @@
...
     }
 
-    // increment1 declares the function to accept its own copy of
-    // and integer value.
+    // increment1 объявляет функцию, создающую свою собственной копию
+    // целочисленного значения.
     func increment1(inc int) {
 
-        // Increment the local copy of the caller’s int value.
+        // Увеличить локальную копию целочисленного значения.
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:93]:
```suggestion
        // Увеличить локальную копию целочисленного значения из вызывающей функции.
```

File: _content/tour/rus/pointers.article [hunk 12]
```diff
@@ -1,151 +1,150 @@
...
     count:  Value Of[ 11 ]  Addr Of[ 0xc000050738 ]
 
-** Notes
+** Заметки
 
-- Use pointers to share data.
-- Values in Go are always pass by value.
-- "Value of", what's in the box. "Address of" ( & ), where is the box.
-- The ( * ) operator declares a pointer variable and the "Value that the pointer points to".
+- Используйте указатели для обмена данными.
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:121]:
```suggestion
- Используйте указатели для совместного использования данных.
```

File: _content/tour/rus/pointers.article [hunk 13]
```diff
@@ -1,151 +1,150 @@
...
+- Значения в Go всегда передаются по значению.
+- "Значение", то, что находится в коробке. "Адрес" ( & ), это где находится коробка.
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:123]:
```suggestion
- "Значение"- это то, что находится в коробке. "Адрес" ( & )- это то, где находится коробка.
```

File: _content/tour/rus/pointers.article [hunk 14]
```diff
@@ -1,151 +1,150 @@
...
-** Escape Analysis
+** Анализ побега (памяти со стэка на кучу)
 
-The algorithm the compiler uses to determine if a value should be constructed
-on the stack or heap is called "escape analysis". The name of the algorithm makes
-it sound like values are constructed on the stack first and then escape (or move)
-to the heap when necessary. This is NOT the case. The construction of a value only
-happens once, and the escape analysis algorithm decides where that will be (stack
-or heap). Only construction on the heap is called an allocation in Go.
+Алгоритм, который компилятор использует для определения, следует ли создавать значение
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:128]:
```suggestion
Алгоритм, который компилятор использует для определения, следует ли выделять значение
```

File: _content/tour/rus/_en_ru_dictionary.yml [hunk 15]
```diff
@@ -38,9 +38,18 @@ Data-oriented design:
 Debugging:
     ru: отладка программного обеспечения
 
+Escape analysis:
+    ru: анализ побега (памяти из стэка на кучу)
```
Comment (rumyantseva) [inline _content/tour/rus/_en_ru_dictionary.yml:42]: I'm not sure this is really used :) I think, usually people call it "escape-анализ", I also heard "анализ утечек", but not "анализ побега". I think, "escape анализ" is the most common. When we mention it for the first time in the article, I suggest using smth like "escape-анализ (анализ утечек)" and then only one of them below.

File: _content/tour/rus/pointers.article [hunk 16]
```diff
@@ -183,154 +182,154 @@ No allocation.
...
 
-** Exercise 2
+** Упражнение 2:
 
-Declare a struct type and create a value of this type. Declare a function that
-can change the value of some field in this struct type. Display the value before
-and after the call to your function.
+Объявите тип структуры и создайте значение этого типа. Объявите функцию, которая
+может изменить значение какого-либо поля в этом типе структуры. Выведите значение до
+и после вызова вашей функции.
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:332]:
```suggestion
Объявите тип struct и создайте значение этого типа. Объявите функцию, которая
может изменить значение какого-либо поля в этой структуре. Выведите значение до
и после вызова вашей функции.
```

File: _content/tour/rus/pointers.article [hunk 17]
```diff
@@ -183,154 +182,154 @@ No allocation.
...
-on the stack, it must happen on the heap. Yes allocation.
+Когда функция escapeToHeap вызывается и завершает выполнение, значение user, которое она создает,
+все равно должно существовать, так как вызывающая функция получает общий доступ к значению.
+Поэтому создание значения внутри функции escapeToHeap не может происходить в стеке,
+оно должно происходить в куче. Да, это выделение памяти в куче.
 
-Think about what would happen if the user value in the last example was constructed
-on the stack when using pointer semantics on the return.
+Подумайте о том, что произойдет, если значение user в последнем примере произойдет в стеке
+используя семантику указателя при возврате.
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:195]:
```suggestion
Подумайте о том, что произойдет, если значение user в последнем примере будет выделено на стеке
используя семантику указателя при возврате.
```

File: _content/tour/rus/pointers.article [hunk 18]
```diff
@@ -183,154 +182,154 @@ No allocation.
...
-if the compiler doesn’t know the size of a value at compile time, the value must be
-constructed on the heap. An example of this is using the built-in function make to
-construct a slice whose size is based on a variable.
+Размер фрейма для каждой функции вычисляется на этапе компиляции. Это означает,
+что если компилятор не знает размер значения на этапе компиляции, значение должно
+быть создано в куче. Примером может послужить использование встроенной функции make
+для создания среза, размер которого зависит от переменной.
 
-    b := make([]byte, size) // Backing array allocates on the heap.
+    b := make([]byte, size) // Массив создается в куче.
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:257]:
```suggestion
    b := make([]byte, size) // Массив для среза создается в куче.
```

File: _content/tour/rus/pointers.article [hunk 19]
```diff
@@ -183,154 +182,154 @@ No allocation.
...
 
-Go uses a contiguous stack implementation to determine how stacks grow and shrink.
-One alternative Go could have used is a segmented stack implementation, which is
-used by some operating systems.
+Go использует вариант непрерывного стек, чтобы определить, как стеки растут и уменьшаются.
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:259]:
```suggestion
Go использует реализацию непрерывного стека, чтобы определить, как стеки растут и уменьшаются.
```

File: _content/tour/rus/pointers.article [hunk 20]
```diff
@@ -183,154 +182,154 @@ No allocation.
...
+Одной из альтернатив, которую мог бы использовать Go - реализация сегментированного стека,
```
Comment (rumyantseva) [inline _content/tour/rus/pointers.article:260]:
```suggestion
Одна из альтернатив, которую мог бы использовать Go - реализация сегментированного стека,
```

**Review by rumyantseva — COMMENTED** [review]
Added some ideas/suggestions.

File: _content/tour/rus/pointers.article (hunk 14 above)
Comment (koleaby4) [reply _content/tour/rus/pointers.article:128]: @rumyantseva , глянь пожалуйста ещё разок - действительно ли `выделять значение` звучит более понятно чем `создавать значение`

Comment (rumyantseva) [reply _content/tour/rus/pointers.article:128]: Давай оставим "создавать", да!

**Review by rumyantseva — APPROVED** [review]
LGTM!

**Pull request merged**